### PR TITLE
#77: implemented SELECT FROM (SELECT ...)

### LINKED ISSUE #77: Add support for SELECT FROM (SELECT ...) queries
Add support for selecting from sub selects.

For example:
```SELECT * FROM (SELECT * FROM ...)```

Until now the sql-statement-builder has no support for adding subselcets to the from clause.

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #77

### CODE REVIEW

File: doc/changes/changes-4.0.0.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# SQL Statement Builder 4.0.0, released ??.06.2020
+# SQL Statement Builder 4.0.0, released 02.07.2020
```
Comment (redcatbear) [inline doc/changes/changes-4.0.0.md:1]:
```suggestion
# SQL Statement Builder 4.0.0, released 2020-07-02
```

File: src/main/java/com/exasol/sql/dql/select/FromClause.java [hunk 2]
```diff
@@ -167,9 +168,32 @@ public FromClause fullOuterJoin(final String name, final String specification) {
         return this;
     }
 
+    /**
+     * Add a select to the {@link FromClause}.
+     *
+     * @param select select statement
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/dql/select/FromClause.java:174]:
```suggestion
     * @param select {@code SELECT} statement
```

Comment (redcatbear) [reply src/main/java/com/exasol/sql/dql/select/FromClause.java:174]: Also in other places

File: src/main/java/com/exasol/sql/dql/select/FromClause.java [hunk 3]
```diff
@@ -167,9 +168,32 @@ public FromClause fullOuterJoin(final String name, final String specification) {
...
+     * @return true if the {@link FromClause} contains an inner select statement
+     */
+    public boolean hasInnerSelect() {
+        return innerSelect != null;
+    }
+
     @Override
     public void accept(final SelectVisitor visitor) {
         visitor.visit(this);
+        if (this.innerSelect != null) {
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/dql/select/FromClause.java:194]:
```suggestion
        if (hasInnerSelect()) {
```

File: src/main/java/com/exasol/sql/dql/select/FromClause.java [hunk 4]
```diff
@@ -179,5 +203,6 @@ public void accept(final SelectVisitor visitor) {
         for (final ValueTable valueTable : this.valueTables) {
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/dql/select/FromClause.java:203]: Combining a sub-Select and a value table is illegal, so it should end in an exception. Also add a test that it does.

File: src/main/java/com/exasol/sql/dql/select/FromClause.java [hunk 5]
```diff
@@ -12,6 +12,7 @@
     private final List<Table> tables = new ArrayList<>();
     private final List<Join> joins = new ArrayList<>();
     private final List<ValueTable> valueTables = new ArrayList<>();
+    private Select innerSelect;
```
Comment (redcatbear) [inline src/main/java/com/exasol/sql/dql/select/FromClause.java:15]:
```suggestion
    private Select subSelect;
```

File: src/main/java/com/exasol/sql/dql/select/FromClause.java (hunk 4 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/sql/dql/select/FromClause.java:203]: done

File: doc/changes/changes-4.0.0.md [hunk 6]
```diff
@@ -1,4 +1,4 @@
-# SQL Statement Builder 4.0.0, released ??.06.2020
+# SQL Statement Builder 4.0.0, released 2020-07-03
```
Comment (redcatbear) [inline doc/changes/changes-4.0.0.md:1]: Update release date.

File: doc/changes/changes-4.0.0.md [hunk 7]
```diff
@@ -9,6 +9,7 @@
 * #80: Added ossindex-maven-plugin and versions-maven-plugin, updated dependencies.
 * #81: Ported from Java 8 to Java 11.
 * #76: Added `SELECT FROM VALUES ... AS` support.
+* #77: Added `SELECT FROM (SELECT ...)`  support.
```
Comment (redcatbear) [inline doc/changes/changes-4.0.0.md:12]:
```suggestion
* #77: Added `SELECT FROM (SELECT ...)` support.
```

File: doc/changes/changes-4.0.0.md (hunk 6 above)
Comment (AnastasiiaSergienko) [reply doc/changes/changes-4.0.0.md]: done

**Pull request merged**